lib/solid_errors/subscriber.rb

```diff
@@ -32,7 +32,7 @@ def report(error, handled:, severity:, context:, source: nil)
       }
       fingerprint = Digest::SHA256.hexdigest(error_attributes.values.join)
       if (record = SolidErrors::Error.find_by(fingerprint: fingerprint))
-        record.update!(resolved_at: nil, updated_at: Time.now)
+        record.update!(prev_resolved_at: record.resolved_at, resolved_at: nil, updated_at: Time.now)
```

Comment:
I do like this because we don't need `prev_resolved_at` set on the first occurrence of an error because the `return true if error.occurrences.count == 1` guard clause deals with that case for us.

Reply:
Awesome!